packages/data/src/components/use-resolve-select/index.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * Internal dependencies
+ */
+import useSelect from '../use-select';
+
+/** @typedef {import('../../types').StoreDescriptor} StoreDescriptor */
+
+/**
+ * Like useSelect, but the selectors return objects containing
+ * both the original data AND the resolution info.
+ *
+ * @param {Function|StoreDescriptor|string} mapSelect see useSelect
+ * @param {Array}                           deps      see useSelect
+ * @return {Function} A custom react hook.
+ */
+export default function useResolveSelect( mapSelect, deps ) {
+	return useSelect( ( select, registry ) => {
+		// Memoization would be nice here
+		const resolve = ( store ) => enrichSelectors( select( store ) );
+		return mapSelect( resolve, select, registry );
+	}, deps );
+}
+
+/**
+ * Transform simple selectors into ones that return an object with the
+ * original return value AND the resolution info.
+ *
+ * @param {Object} selectors Selectors to enrich
+ * @return {Object} Enriched selectors
+ */
+function enrichSelectors( selectors ) {
+	const resolvers = {};
+	for ( const selectorName in selectors ) {
```

Comment:
If you map all fields of `selectors` like this, won't it also map `getIsResolving`, `has{Started,Finished}Resolution` and other meta-selectors? We don't want that I guess.